src/watson/models.py

```diff
@@ -70,14 +72,21 @@ class SearchEntry(models.Model):
 
     meta_encoded = models.TextField()
 
-    @property
+    def _deserialize_meta(self):
+        from watson.registration import SearchEngine
+        engine = SearchEngine._created_engines[self.engine_slug]
+        model = self.content_type.model_class()
```

Comment:
it would be nice if there was some way to detect that the adapter was just using vanilla `json.loads(...)` and avoid doing this query, so existing users don't get a sudden slow down from 1 query per `SearchEntry`